caddyhttp/httpserver/server.go

```diff
@@ -126,15 +126,16 @@ func (s *Server) Listen() (net.Listener, error) {
 		ln = tcpKeepAliveListener{TCPListener: tcpLn}
 	}
 
+	cln := ln.(caddy.Listener)
```

Comment:
Oh yeah, duh, because the middleware wrapper takes in a caddy.Listener too. :+1: